lib/improver/spotdata/common_functions.py

```diff
@@ -0,0 +1,240 @@
+    root_no_neighbours = np.sqrt(no_neighbours)
+    delta_neighbours = (root_no_neighbours - 1)/2
+    if not np.mod(delta_neighbours, 1) == 0 or delta_neighbours < 1:
+        raise ValueError(
+            'Invalid neareat no. of neighbours request. N={} is not a valid '
+            'square no. (sqrt N must be odd)'.format(no_neighbours))
+
+    delta_neighbours = int(delta_neighbours)
+    n_indices = [(i+a, j+b)
+                 for a in range(-delta_neighbours, delta_neighbours+1)
+                 for b in range(-delta_neighbours, delta_neighbours+1)]
+    if exclude_self is True:
+        n_indices.pop(no_neighbours/2)
+    return np.array(
+        [np.array(n_indices)[:, 0], np.array(n_indices)[:, 1]]
+        ).astype(int).tolist()
+
+
+def node_edge_test(node_list, cube):
+    '''
+    Node lists produced using the nearest_n_neighbours function may overspill
+    the domain of the array from which data is to be extracted. This function
+    checks whether the cube of interest is a global domain with a wrapped
+    boundary using the iris.cube.Cube.coord().circular property. In cases of
+    wrapped boundaries, the neighbouring points addresses are appropriately
+    modified. Otherwise the points are discarded.
+
+    Args
+    ----
+    node_list : list[[i],[j]] of indices.
+    cube      : the cube for which data will be extracted using the
+                indices (e.g. cube.data[node_list]).
+
+    Returns
+    -------
+    node_list : modified node_list with points beyond the cube boundary
+                either changed or discarded as appropriate.
+
+    '''
+
+    node_list = np.array(node_list)
+
+    for k, coord in enumerate(['y', 'x']):
+        coord_max = cube.coord(axis=coord).shape[0]
+        circular = cube.coord(axis=coord).circular
+        max_list = np.where(node_list[k] >= coord_max)[0].tolist()
+        min_list = np.where(node_list[k] < 0)[0].tolist()
+        if circular:
+            node_list[k, min_list] = node_list[k, min_list] + coord_max
+            node_list[k, max_list] = node_list[k, max_list] - coord_max
+        else:
+            node_list = np.delete(node_list,
+                                  np.hstack((min_list, max_list)), 1)
+
+    return node_list.tolist()
+
+
+def get_nearest_coords(cube, latitude, longitude):
+    '''
+    Uses the iris cube method nearest_neighbour_index to find the nearest grid
+    points to a given latitude-longitude position.
+    '''
+
+    i_latitude = (cube.coord(axis='y').nearest_neighbour_index(latitude))
+    j_longitude = (cube.coord(axis='x').nearest_neighbour_index(longitude))
+    return i_latitude, j_longitude
+
+
+def index_of_minimum_difference(whole_list, subset_list=None):
+    '''
+    Returns the index of the minimum value in a list.
+    '''
+    if subset_list is None:
+        subset_list = np.arange(len(whole_list))
+    return subset_list[np.argmin(abs(whole_list[subset_list]))]
+
+
+def list_entry_from_index(list_in, index_in):
```

Comment:
Used the second of your methods, but have left it inside a function as I will remember the name of the function much more easily than reproducing this code.